sound/soc/sof/pm.c

```diff
@@ -340,7 +340,10 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 #endif
 	/* notify DSP of upcoming power down */
 	ret = sof_send_pm_ipc(sdev, SOF_IPC_PM_CTX_SAVE);
-	if (ret < 0) {
+	if (ret == -ETIMEDOUT) {
+		dev_warn(sdev->dev,
+			 "warn: ctx_save ipc timeout, forced suspend\n");
+	} else if (ret < 0) {
```

Comment:
question: why do we prevent suspend on *any* IPC error on CTX_SAVE? @ranj063?

Reply:
@plbossart in the past we did let the DSP suspend even in the case of errors. But more recently we've fixed the power down sequence in the FW and so this should technically never fail.

Reply:
But coming back to this patch, if the DSP crashed even before the CTX_SAVE is sent, why do we even send out the ipc?

Reply:
I think we should come up with a more intelligent way to tell that the DSP has crashed (based on a previous ipc timeout) and not this ipc in the first place to allow for the dsp to be powered down.

Reply:
> @plbossart in the past we did let the DSP suspend even in the case of errors. But more recently we've fixed the power down sequence in the FW and so this should technically never fail.

Humm, I am old enough to question these statements. Murphy's Law applies. If there is by-design no error then let's remove it. If not let's document what the expected errors might bet and leave room for the 'known unknowns'

Reply:
Thanks @ranj063 and @plbossart for the comments!

I'll change this so that we do not prevent suspend on any errors. This would indeed seem like the right thing to do.